Wpb 10872 | Update docs to always deploy RabbitMQ and add data persistence (#730)
wireapp/wire-server-deploy: offline/rabbitmq_setup.md

Local host files (e.g. /etc/hosts)
Reference - https://www.rabbitmq.com/clustering.html#cluster-formation-requirements

For adding entries to local host file(`/etc/hosts`), run
```
d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/roles/rabbitmq-cluster/tasks/configure_dns.yml
```

Create the rabbitmq cluster:

``` 
d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/rabbitmq.yml
```

and run the following playbook to create values file for helm charts to look for RabbitMQ IP addresses -

```
d ansible-playbook -i ./ansible/inventory/offline/hosts.ini ansible/helm_external.yml --tags=rabbitmq-external
```

Make Kubernetes aware of where RabbitMQ external stateful service is running:
```
d helm install rabbitmq-external ./charts/rabbitmq-external --values ./values/rabbitmq-external/values.yaml
```

Configure wire-server to use the external RabbitMQ service:

Edit the `/values/wire-server/prod-values.example.yaml` file to update the RabbitMQ host
Under `brig` and `galley` section, you will find the `rabbitmq` config, update the host to `rabbitmq-external`, it should look like this:
```
rabbitmq:
  host: rabbitmq-external
``` 

## Backup and Restore

> Following steps describe the backup and restore process for RabbitMQ deployed outside of Kubernetes.

Reviewer: ```suggestion
The following steps describe the backup and restore process for RabbitMQ deployed outside of Kubernetes.
```

> Although, this can vary based on your setup, it is also recommended to follow the official documentation here - https://www.rabbitmq.com/docs/backup

Reviewer: ```suggestion
This can vary based on your security, privacy, and administrative policies. It is also recommended to read and follow the official documentation here - https://www.rabbitmq.com/docs/backup
```

## Backup
Make sure to have the nodes on which RabbitMQ is running in the [ansible inventory file](https://github.com/wireapp/wire-server-deploy/blob/master/offline/docs_ubuntu_22.04.md#editing-the-inventory), under the `rmq-cluster` group.

> Then run the following command:

Reviewer: ```suggestion
Then run the following command to load your wire utility environment:
```

Now, update the `./values/rabbitmq/values.yaml` and `./values/rabbitmq/secrets.yaml` with correct values as needed.

Deploy the `rabbitmq` helm chart:
```
d helm upgrade --install rabbitmq ./charts/rabbitmq --values ./values/rabbitmq/values.yaml --values ./values/rabbitmq/secrets.yaml
```

### Method 2: Install RabbitMQ outside of the Kubernetes cluster with an Ansible playbook

Add the nodes on which you want to run rabbitmq to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbitmq-cluster/defaults/main.yml` file with the correct configurations for your environment.

If you need RabbitMQ to listen on a different interface than the default gateway, set `rabbitmq_network_interface`

You should have following entries in the `/ansible/inventory/offline/hosts.ini` file. For example:
```
[rmq-cluster:vars]
rabbitmq_network_interface = enp1s0

[rmq-cluster]
ansnode1
ansnode2
ansnode3
```

> 

Reviewer: ```suggestion
```
